examples/server/server.cpp

```diff
@@ -3148,6 +3176,84 @@ int main(int argc, char ** argv) {
         res.status = 200; // HTTP OK
     };
 
+    const auto handle_get_control_vectors = [&ctx_server](const httplib::Request & req, httplib::Response & res) {
+      json vectors = json::array();
+
+      for (const auto & vec : ctx_server.params.control_vectors) {
+          vectors.push_back(json {
+              { "fname", vec.fname },
+              { "strength", vec.strength }
+          });
+      }
+      json data = {
+          { "vectors", vectors },
+          { "layer_start", ctx_server.params.control_vector_layer_start },
+          { "layer_end", ctx_server.params.control_vector_layer_end }
+      };
+      res.set_content(data.dump(), "application/json; charset=utf-8");
+    };
+
+    const auto handle_set_control_vectors = [&ctx_server, &res_error, &handle_get_control_vectors](const httplib::Request & req, httplib::Response & res) {
+        res.set_header("Access-Control-Allow-Origin", req.get_header_value("Origin"));
+
+        json data = json::parse(req.body);
+        std::vector<llama_control_vector_load_info> vec_params;
+
+        if (data.contains("vectors") && data["vectors"].is_array()) {
+            for (const auto &item : data["vectors"]) {
+                auto v = item.get<llama_control_vector_load_info>();
+                std::cout << "Add vector: " << v.fname << " " << v.strength << "\n";
+                vec_params.push_back(v);
+            }
+        } else {
+            std::cerr << "No vectors passed\n";
+            res_error(res, format_error_response("No vectors passed", ERROR_TYPE_SERVER));
+            return;
+        }
+        const auto cvec = llama_control_vector_load(vec_params);
```

Comment:
I would prefer the server to be restarted than hot loading anything from a user input file.

Reply:
Swapping control vectors live was my main motivation. I'd really like to use them like samplers, without reloading the model when I change them. But I can't figure out how the low-level stuff works and why my naive approach produces garbage. So I'll just transform this now into the trivial "load control vectors on start" PR, which just copies the functionality from main.

But I would really like to be able to swap them live and would be thrilled if someone found out how that might work.